app/Http/Controllers/RoomController.php

```diff
@@ -136,54 +128,39 @@ public function store(Request $request)
         $playerRoom->room()->associate($room);
 		$playerRoom->save();
 
-        // TODO: In style with the other APIs we should return an empty 201 response
-        // with a Location header to the room.
-        return new Response([
-            'created_room' => [
-                'room_id' => $room->id,
-                'creator' => $this->session->player->id,
-                'description' => $room->description,
-                'rated' => (bool)$room->is_rated,
-                'max_players' => $room->max_players,
-                'player_count' => 1, // TODO: This is always only the creator, do we really need to return this?
-                'min_rating' => $room->min_rating,
-                'goal' => $room->goal->identifier,
-                'anonymity' => (bool)$room->is_anonymous,
-                'map' => $room->map,
-                'seed' => $room->seed,
-            ],
-        ]);
+        return new CreatedResponse($room);
     }
 
     /**
      * Show a room.
      *
-     * @param int $roomId
+     * @param Room $room
      * @return Response
      */
-    public function show(int $roomId)
+    public function show(Room $room)
     {
-        if (!$room = Room::whereId($roomId)->first()) {
-            return new NotFoundResponse();
-        }
-
         return new Response($room);
     }
 
     /**
      * Update a room.
      *
-     * @param $roomId
+     * @param Room $room
      * @param Request $request
      * @return Response
      * @throws ValidationException
      */
-    public function update($roomId, Request $request)
+    public function update(Room $room, Request $request)
     {
-        $this->validator($request)->validate();
-        if (!$room = Room::whereId($roomId)->first()) {
-            return new NotFoundResponse();
-        }
+        $request->validate([
+            'max_players' => 'required|int|between:2,10',
+            'goal' => 'required|string|' . Rule::in(Goal::pluck('identifier')),
+            'description' => 'required|string',
+            'map' => 'required|int|between:0,3',
+            'min_rating' => 'required|int|min:0', // TODO: Make this  for open games?
```

Comment:
I don't think casual games should have rating limits.

Reply:
I might not be understanding correctly, but I do think players should be able to impose rating limits on casual games. Otherwise couldn't anyone just join any casual game? I think that would make it not as fun for high rated players looking to play a casual game with others of similar skill.

Reply:
I'm not sure, can you set rating limits in casual games in OG subs? I don't remember

Reply:
You can't, but I definitely think you should be able to. I don't think you should be forced to play rated just to get a decent game. Plus there won't necessarily be a rating system for every mode of play, so you might want to play a custom game but with people who are good. In that case you wouldn't be able to unless you could set a rating restriction on casual games.

Reply:
I agree with Tony, it's pretty reasonable to request having players in a certain skill range even for casual games. Imagine a hard-core group of pro players going casual, getting matched with newbies who're just looking for a simple game, and the noobs get crushed.

Imagine combining that with a "you need to play 10 games before going ranked" idea and you might have a large fraction of newbies quitting the game early...

However, I think we should make the `min_rating` argument optional in case the user does not want to set a minimum. Same thing for the currently non-existing `max_rating`.

Reply:
This sounds good to me! I will close this one and we can leave it as is